### ISSUE TITLE
Question about Autotracker

### ISSUE THREAD

**Issue by rifton007** [toggl/toggldesktop#2231]
Questions about Autotracker

When i create a new application in the Autotracker,
why must i fill the name of projet ?
Because, in my case i do not use Adobe InDesign for a unique project, but well for many projects.

My second question is:
Is it possible for the Autotracker to detect the closing of a applicaiton and stop the current timer ?

Thanks for you answers.

**Comment by IndrekV** [comment]
1. Currently the autotracker work only with projects, but I see what you are saying that with certain apps the project is not always the same.

2. The autotracer does not know which applications are running it only knows the application in focus so not it cannot track when application is closed.

**Comment by rifton007** [comment]
1. That's right, and I'm trying to find a way to use the autotracker effectively.

Let's take a concrete example.

- I have a project to create a logo for the company A.
- Another project to create a flyer, also for company A.
- And in three days, another logo design for company B.

For all these different projects and for different clients, I have to use the same application, Adobe Illustrator. 
Toggl is a great tool, but I do not quite understand how to use autotracking in my work.

2. I never programmed in C ++. But this reasoning is possible in C ++ ?

1. Toogle detects the focus of a window.
2. Toogle displays a popup to the user to start tracking.
3. The user clicks "START"
4. Toogle starts the timer and saves the PID of that window.
5. Toogle listens to the process with its PID, if the process stops Toggle stops the timer.

Thank for your time.

**Comment by IndrekV** [comment]
The PID monitoring is not currently implemented. Currently only focused window is monitored and the autotracker and timeline data is based on this info.

**Comment by muyncky** [comment]
Can I suggest something too?

For a programmer who opens files in their text editor (Sublime Text, Notepad++) it would be handy to link certain file paths to projects. Same goes for the designer examples above. Let's say:
"upon opening a file in Finder/Explorer with file path x project y should be started." I don't know if the app with which it is opened then is still relevant  though...
A match could be anything inside a certain folder. It is only a matter of mapping the right folders.

**Comment by IndrekV** [comment]
@muyncky The idea of monitoring files opened by user seems very intrusive. The way it would work is if the app would hold the location of filename in their window title. Autotracker reads window titles to see what app is opened and if window title has any info about the project or file that is opened it could be used as a matching term in autotracker settings.

**Comment by muyncky** [comment]
@IndrekV that is true! And I see now, it works. Sublime Text does have the pathname in the title! I think I configured that one day.

Thanks!
